src/views/FlawSearchView.vue

```diff
@@ -1,67 +1,82 @@
 <script setup lang="ts">
 import IssueSearchAdvanced from '@/components/IssueSearchAdvanced.vue';
 import { z } from 'zod';
-import { searchFlaws } from '../services/FlawService';
-import { onMounted, watch, ref } from 'vue';
+import { computed, ref, watch } from 'vue';
 import { useRoute } from 'vue-router';
 import IssueQueue from '@/components/IssueQueue.vue';
+import { useFlaws }  from '../composables/useFlaws';
+
+const { issues, isLoading, isFinalPageFetched, loadFlaws, loadMoreFlaws } = useFlaws();
 
 const route = useRoute();
 
+const filters = ref<Record<string, string>>({});
+const tableFilters = ref<Record<string, string>>({});
+
 defineProps<{
   query: string;
 }>();
 defineEmits<{
   'issues:load': [any[]];
 }>();
-const issues = ref<any[]>([]);
-const isLoading = ref(false);
 
 const searchQuery = z.object({
   query: z.object({
-    query: z.string(),
+    query: z.string().nullish(),
   }),
 });
 
-function search() {
-  try {
-    const parsedRoute = searchQuery.parse(route);
-    if (parsedRoute.query.query === '') {
-      // TODO handle error
-      return;
-    }
-    isLoading.value = true;
-    searchFlaws(parsedRoute.query.query)
-      .then((response) => {
-        console.log('IssueSearch: got flaws: ', response.data);
-        issues.value = response.results;
-      })
-      .catch((err) => {
-        console.error('IssueSearch: getFlaws error: ', err);
-      })
-      .finally(() => {
-        isLoading.value = false;
-      });
-  } catch (e) {
-    console.log('IssueSearch: error advanced searching', e);
-    isLoading.value = false;
+const params = computed(() => {
+  const parsedRoute = searchQuery.parse(route);
+  const paramsObj = {
+    ...filters.value,
+    ...tableFilters.value
+  };
+  if (parsedRoute.query.query) {
+    paramsObj.search = parsedRoute.query.query;
   }
+  return paramsObj;
+});
+
+watch(() => params, () => {
+  loadFlaws(params);
+}, {
+  deep: true
+});
+
+function fetchMoreFlaws() {
+  loadMoreFlaws(params);
 }
 
-onMounted(search);
-watch(() => route.query?.query, search);
+function setFilters(newFilters : Record<string, string>) {
+  filters.value = {
+    ...newFilters
+  };
+}
 
-function setIssues(loadedIssues: any[]) {
-  issues.value = loadedIssues;
+function setTableFilters(newFilters: any = {}) {
+  tableFilters.value = {
+    ...newFilters.value
+  };
 }
```

Comment:
What is the difference between the filter and table filter here? :thinking:

Reply:
`filter` is from AdvancedSearch filter.
`tableFilter` is myIssue and active issue filter on the table.

![image](https://github.com/RedHatProductSecurity/osim/assets/156366006/1a54164c-e0d8-4c80-9804-d8ac548b896f)